test/jdk/javax/accessibility/JSlider/AccessibleAction/JSliderAccessibleAction.java

```diff
@@ -0,0 +1,286 @@
+                            "changed when invalid decrement value -1 is passed to " +
+                            "doAccessibleAction(-1)  jSliderInitialValue = "
+                            + jSliderInitialValue + "  currentJSliderValue = " + currentJSliderValue);
+                }
+
+                // JSlider value is decremented
+                mouseAction(robot, decrementBtn);
+                if (!validDecrementCountDownLatch.await(30, TimeUnit.SECONDS)) {
+                    throw new RuntimeException("Failed to perform action on valid " +
+                            "decrement button");
+                }
+                if (jSliderInitialValue.get() == currentJSliderValue.get()) {
+                    throw new RuntimeException("Expected that JSlider value is  " +
+                            "decremented when value 1 is passed to " +
+                            "doAccessibleAction(1)  jSliderInitialValue = "
+                            + jSliderInitialValue + "  currentJSliderValue = " + currentJSliderValue);
+                }
+
+                // JSlider value is incremented
+                mouseAction(robot, incrementBtn);
+                if (!validIncrementCountDownLatch.await(30, TimeUnit.SECONDS)) {
+                    throw new RuntimeException("Failed to perform action on valid " +
+                            "Increment button");
+                }
+                if (jSliderInitialValue.get() != currentJSliderValue.get()) {
+                    throw new RuntimeException("Expected that JSlider value is  " +
+                            "incremented when value 0 is passed to " +
+                            "doAccessibleAction(0)  jSliderInitialValue = "
+                            + jSliderInitialValue + "  currentJSliderValue = " + currentJSliderValue);
+                }
+            } finally {
+                SwingUtilities.invokeAndWait(() -> {
+                    if (jFrame != null) {
+                        jFrame.dispose();
+                    }
+                });
+            }
+        }
+    }
+
+    public static void mouseAction(Robot robot, JButton button) throws InterruptedException,
+            InvocationTargetException {
+        robot.waitForIdle();
+        final Point[] point = new Point[1];
+        final Rectangle[] rect = new Rectangle[1];
+
+        SwingUtilities.invokeAndWait(() -> {
+            point[0] = button.getLocationOnScreen();
+            rect[0] = button.getBounds();
+        });
+
+        robot.mouseMove(point[0].x + rect[0].width / 2,
+                point[0].y + rect[0].height / 2);
+        robot.waitForIdle();
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        robot.waitForIdle();
+    }
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException, AWTException {
+        JSliderAccessibleAction jSliderAccessibleAction =
```

Comment:
The class appears to have no state, why do you need an instance of it? Why not to make the methods static?

Reply:
Fixed make the method to be static